internal/providers/msentraid/msentraid.go

```diff
@@ -273,12 +273,17 @@ func (p Provider) CurrentAuthenticationModesOffered(
 	return offeredModes, nil
 }
 
+// NormalizeUsername parses a username into a normalized version.
+func (p Provider) NormalizeUsername(username string) string {
+	return strings.ToLower(username)
```

Comment:
Is this a new logic defined somewhere?

I think it's correct, but it wasn't used before, was it?

Reply:
It's defined in the latest version of the [spec](https://docs.google.com/document/d/1GK2yZ71_w2WxyNndap47RheezKGNCC4RyElKKeAmHL0/edit?tab=t.0#heading=h.azt725jx5rb8). Before `strings.EqualFold` was used which is case insensitive.

The reason for adding this method is that we wish to check if the normalized username is in the `allowed_users` list, so we had to extract the normalization logic in a separate function.